### PR TITLE
Handle circled grid squares in puzzle

### PR DESCRIPTION
**Pull request by jkboyce**
Detects if there are any marked up (circled) squares in the puzzle and adds them to the output file.

Recent example:
`nyt.py Chrome https://www.nytimes.com/crosswords/game/daily/2021/09/12 Sep1221.puz`

### CODE REVIEW

File: nyt.py [hunk 1]
```diff
@@ -272,6 +272,15 @@ def data_to_puz(puzzle):
             else:
                 rebus.add_rebus(None)
 
+    # See if any grid squares are marked up with circles
+    if any(x['type'] == 2 for x in data['cells'] if 'type' in x):
```
Comment (edsantiago) [inline nyt.py:276]: Use of magic constant `2`. Would you consider refactoring this to a named constant such as `NYT_TYPE_CIRCLED`? That would make it easier to search for other uses (such as line 281), and to change/edit/extend as necessary in the future (e.g. if the type turns out to be a bitmask).

**Review by edsantiago — APPROVED** [review]
LGTM. One minor maintainability suggestion. Otherwise, works as advertised. Thank you!

File: nyt.py (hunk 1 above)
Comment (Q726kbXuN) [reply nyt.py:276]: Agreed, I'll make this change after the PR, including all of the possibilities for values

**Comment by jkboyce** [comment]
Thanks for the change @edsantiago and @Q726kbXuN -- looks great!

**Pull request merged**